x-pack/plugins/maps/server/content_management/maps_storage.ts

```diff
@@ -0,0 +1,312 @@
+
+    // Validate input (data & options) & UP transform them to the latest version
+    const { value: dataToLatest, error: dataError } = transforms.update.in.data.up<
+      MapAttributes,
+      MapAttributes
+    >(data);
+    if (dataError) {
+      throw Boom.badRequest(`Invalid data. ${dataError.message}`);
+    }
+
+    const { value: optionsToLatest, error: optionsError } = transforms.update.in.options.up<
+      CreateOptions,
+      CreateOptions
+    >(options);
+    if (optionsError) {
+      throw Boom.badRequest(`Invalid options. ${optionsError.message}`);
+    }
+
+    // Save data in DB
+    const soClient = await savedObjectClientFromRequest(ctx);
+    const partialSavedObject = await soClient.update<MapAttributes>(
+      SO_TYPE,
+      id,
+      dataToLatest,
+      optionsToLatest
+    );
+
+    // Validate DB response and DOWN transform to the request version
+    const { value, error: resultError } = transforms.update.out.result.down<
+      MapUpdateOut,
+      MapUpdateOut
+    >({
+      item: savedObjectToMapItem(partialSavedObject, true),
+    });
+
+    if (resultError) {
+      throw Boom.badRequest(`Invalid response. ${resultError.message}`);
+    }
+
+    return value;
+  }
+
+  async delete(ctx: StorageContext, id: string): Promise<MapDeleteOut> {
+    const soClient = await savedObjectClientFromRequest(ctx);
+    await soClient.delete(SO_TYPE, id);
+    return { success: true };
+  }
+
+  async search(
+    ctx: StorageContext,
+    query: SearchQuery,
+    options: MapSearchOptions = {}
+  ): Promise<MapSearchOut> {
+    const {
+      utils: { getTransforms },
+      version: { request: requestVersion },
+    } = ctx;
+    const transforms = getTransforms(cmServicesDefinition, requestVersion);
+    const soClient = await savedObjectClientFromRequest(ctx);
+
+    // Validate and UP transform the options
+    const { value: optionsToLatest, error: optionsError } = transforms.search.in.options.up<
+      MapSearchOptions,
+      MapSearchOptions
+    >(options);
+    if (optionsError) {
+      throw Boom.badRequest(`Invalid payload. ${optionsError.message}`);
+    }
+    const { onlyTitle = false } = optionsToLatest;
+
+    const { included, excluded } = query.tags ?? {};
+    const hasReference: SavedObjectsFindOptions['hasReference'] = included
+      ? included.map((id) => ({
+          id,
+          type: 'tag',
+        }))
+      : undefined;
+
+    const hasNoReference: SavedObjectsFindOptions['hasNoReference'] = excluded
+      ? excluded.map((id) => ({
+          id,
+          type: 'tag',
+        }))
+      : undefined;
+
+    const soQuery: SavedObjectsFindOptions = {
+      type: CONTENT_ID,
+      search: query.text,
+      perPage: query.limit,
+      page: query.cursor ? +query.cursor : undefined,
+      defaultSearchOperator: 'AND',
+      searchFields: onlyTitle ? ['title'] : ['title^3', 'description'],
+      hasReference,
```

Comment:
`fields: ['description', 'title'],` should be retained (revert https://github.com/elastic/kibana/commit/0725ebbe53920d3c3b6eb2ed474648ea030bd64f).

Listing page does not use real pagination and fetches advanced setting `savedObjects:listingLimit` objects. This value defaults to 1000, https://github.com/elastic/kibana/blob/3f60fd17619f95c7a6d2faae5dbb7dd0139010e1/x-pack/plugins/maps/public/routes/list_page/maps_list_view.tsx#L93

Loading all attributes will have large performance implications for customers with a large number of maps saved objects.

I do not understand why this was removed. Can some one explain the reasoning and why they outweigh the application performance issues that it will cause?

Reply:
FWIW data views work the same way but I suspect the performance benefits are marginal.

Reply:
> Can some one explain the reasoning and why they outweigh the application performance issues that it will cause?


The benefit of not using `fields` and returning the whole object is to make content management storage work with one single object shape. Just an exaggerated example: instead of having types like `Map` `MapSummary` `MapWithDescriptionAndTitleOnly` `FullMap` scattered around client/server code, we would have only `Map`. 
This could potentially not only make maps code simpler but also later could bring more benefits for us in content management: If we have constraints that `get` `bulkGet` `create` `search` `msearch` must return the same objects, we should be able to build reacher shared UI components and have a better client caching system on top of CM with minimal changes for the consumers. 


> Loading all attributes will have large performance implications for customers with a large number of maps saved objects.

As far as I understand this shouldn't have large performance implications for customers. Turns out today both Dashboard and Visualize listing pages fetch the whole objects. I assume if they could keep it like this until this point, this shouldn't be breaking for maps. 

> Listing page does not use real pagination and fetches advanced setting savedObjects:listingLimit objects. This value defaults to 1000,

I think we agree that ideally, we should add real pagination. Seems like long-term TableListView will be natively integrated with CM and then we can implement this for you. Then also visualize and dashboard get the same benefit.

Reply:
Ok @Dosant. Let's revisit the need for it when we actually need it 👍 

I reverted the change in https://github.com/elastic/kibana/pull/153304/commits/5adbb6465b5ca7356ce33d8ae2ecce42cd4b3d69